subiquity/server/runner.py

```diff
@@ -49,16 +76,36 @@ def __init__(self, ident, delay):
         super().__init__(ident)
         self.delay = delay
 
-    async def start(self, cmd):
+    def _forge_systemd_cmd(self, cmd: List[str], private_mounts: bool) \
+            -> List[str]:
+        # We would like to use systemd-run here but unfortunately it requires
+        # root privileges.
+        # Using systemd-run --user would be an option but it not available
+        # everywhere ; so we fallback to using systemd-cat.
```

Comment:
where is it not available, out of curiosity?

Reply:
"not available" might not be accurate. My first implementation was using the `--user` switch all the time and it did not work in the installer because it runs as root:
```bash
# systemd-run --user /bin/ls
Failed to connect to bus: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)
```
We can't say "use `--user` when in dry-run mode and not when in normal mode" because sometimes we run integration tests (i.e. dry-run) as root.

Maybe the "if not root, then use `--user` would work" but I don't know ; it feels brittle. What do you think?

Reply:
I think "if not root, then use --user" might actually make sense, even if it is a little strange. The more we can exercise stuff like this in dry-run mode, the better.

Reply:
There we go!